### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by SSEKPIUS**
Add Redux Toolkit(npm install react-redux @reduxjs/toolkit).

Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:

...
/src
    |---/components
    |---/redux
        |--- /books
                | books.js
        |--- /categories
                | categories.js
        | configureStore.js
    | index.js
Write your book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

Define action types for adding and removing a book.
Set the initial state to be an empty array of books.
Export Action Creators for your actions.
Write your reducer and export it as default.
Define state changes for the actions that you created.
In case of unknown action - return the current state.
Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

Define an action type for checking the status.
Set the initial state to be an empty array of categories.
Export Action Creators for your actions.
Write your reducer and export it as default.
For the check status action return a string "Under construction".
In case of unknown action - return the current state.
Configure the Redux Store (/src/redux/configureStore.js):

Importing the necessary methods from Redux Toolkit.
Combine both reducers into a root reducer by using configureStore function.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Actions
+const ADD_BOOK = 'bookstore/books/ADD';
+const REMOVE_BOOK = 'bookstore/books/REMOVE';
```
Comment (danmainah) [inline src/redux/books/books.js]: - [ ] @SSEKPIUS great work on your reducer, however in this case the action type definition doesn't follow the[ ducks pattern](https://github.com/erikras/ducks-modular-redux), the last name in your case 'ADD' should be equal to the const in your case 'ADD_BOOK'

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Actions
+const CHECK_STATUS = 'bookstore/categories/STATUS';
```
Comment (danmainah) [inline src/redux/categories/categories.js]: - [ ] @SSEKPIUS great work on your reducer, however in this case the action type definition doesn't follow the[ ducks pattern](https://github.com/erikras/ducks-modular-redux), the last name in your case 'STATUS' should be equal to the const in your case 'CHECK_STATUS'

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @SSEKPIUS , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Gopxfs — APPROVED** [review]
# 🟢 STATUS: APPROVED 🟢

## Hello @SSEKPIUS 👋,

Your project is complete! You followed all the requirements, feel free to merge it! 😄
Congratulations! 🎉 



## Highlights
✔️ Linters are passing successfully.
✔️ Implemented all the changes requested by the previous reviewer. Well done!

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**